website/client-old/js/controllers/inventoryCtrl.js

```diff
@@ -253,20 +253,36 @@ habitrpg.controller("InventoryCtrl",
             size:'sm'
           });
         }
-
-        // Checks if mountmaster has been reached for the first time
-        if(!user.achievements.mountMaster
-            && $scope.mountCount >= 90) {
-          User.user.achievements.mountMaster = true;
-          Achievement.displayAchievement('mountMaster');
-        }
+        handleMountAchievements();
 
       // Selecting Pet
       } else {
         User.equip({params:{type: 'pet', key: pet.key}});
       }
     }
+    function handleMountAchievements() {
+        //Mount collector
+        var retrospectiveCount = (user.achievements.mountMaster || 0) * 90 + $rootScope.countExists(user.items.mounts);
+        user.achievements.mountCollector = Math.max(retrospectiveCount, user.achievements.mountCollector || 0);
+        switch (user.achievements.mountCollector) {
+            case 1:
+                break;
+            case 10:
+                break;
+            case 25:
+                break;
+            case 50:
+                break;
+        }
 
+        // Checks if mountmaster has been reached for the first time
+        if (!user.achievements.mountMaster
+            && $scope.mountCount >= 90) {
+            User.user.achievements.mountMaster = true;
```

Comment:
So, the issue here is that you are changing the users value on the client. You need to move this code to the server so the user gets saved to the DB. Probably here: https://github.com/HabitRPG/habitica/blob/develop/website/common/script/ops/feed.js#L77 or in the evolve function.